addons/web_editor/views/snippets.xml

```diff
@@ -105,6 +102,62 @@
                 <t t-set="indent" t-value="True"/>
             </t>
         </div>
+
+        <div data-js="BackgroundShape" string="Shape"
+            t-att-data-selector="selector"
+            t-att-data-exclude="exclude"
+            t-att-data-target="target">
+            <we-select-pager class="o_we_shape_menu" string="Shape" data-dependencies="!shape_none_opt" data-name="bg_shape_opt">
+                <!-- Invisible None button to be used for select's placeholder text -->
+                <we-button data-shape="" data-name="shape_none_opt">None</we-button>
+                <we-select-page>
+                    <we-button data-shape="website/shape_bg_1">Shape 1</we-button>
+                    <we-button data-shape="website/shape_bg_2">Shape 2</we-button>
+                    <we-button data-shape="website/shape_bg_3">Shape 3</we-button>
+                    <we-button data-shape="website/shape_bg_4">Shape 4</we-button>
+                    <we-button data-shape="website/shape_bg_5">Shape 5</we-button>
+                    <we-button data-shape="website/shape_bg_6">Shape 6</we-button>
+                    <!-- Would be better to have the x/y flip as a separate data-attribute, but this is
+                    not possible with a current system, as the method is triggred on the select, and method
+                    params are unique per data-method on a given widget -->
+                    <we-button data-shape="website/shape_bg_1,x">Shape 1 flip-x</we-button>
+                    <we-button data-shape="website/shape_bg_2,xy">Shape 2 flip-xy</we-button>
```

Comment:
If I understand your explanation, this is not true anymore since: https://github.com/odoo/odoo/commit/a30cc3402d1ba2678bf2219a33972aff3ef65592

You may probably simplify your code then

Reply:
I would need to extend `widget.setValue` to pass params as well so that I can show the correct shape as active, would that be fine?

Reply:
Pfft :/ No... In fact I am not sure to understand this feature. Do you have shapes which are already flipped in the shape selector? Then if you chose them and flip them... would that be possible to distinguish from the unfliped version? Is this really needed?

Reply:
Might need to talk about this with SBU/designers. There are a few shapes that are the same except they're flipped, but maybe it's ok to duplicate the svg files for these so we don't have this headache.

Currently, since they're the same shape, if a shape is flipped, I try to find a corresponding button in the select with that x/y flip, and if there is none, I set the unflipped version as active. So if you flip the `Shape 1 flip-x`, it will show `Shape 1` as active, and if you flip `Shape 1` on the x axis, it will show `Shape 1 flip-x` as active, it's not super great but it's not awful either

Reply:
Yes that's what I expected by reading your code and it is confusing to me. The code would be easier without this and the functional behaviour too. Let's duplicate those SVG files.